bin/_mocha

```diff
@@ -190,16 +190,41 @@ Error.stackTraceLimit = Infinity; // TODO: config
 
 var reporterOptions = {};
 if (program.reporterOptions !== undefined) {
-    program.reporterOptions.split(",").forEach(function(opt) {
-        var L = opt.split("=");
-        if (L.length > 2 || L.length === 0) {
-            throw new Error("invalid reporter option '" + opt + "'");
-        } else if (L.length === 2) {
-            reporterOptions[L[0]] = L[1];
-        } else {
-            reporterOptions[L[0]] = true;
+  var reporterOptionsParser;
+  var multipleReporterFormat = /([^,:]+):{([^}]+)}/g;
+  if (program.reporterOptions.match(multipleReporterFormat)) {
+    // multiple reporter config:
+    //   spec:{a=1,b=2},dot:{c=3,d=4}
+    var match;
+    while((match = multipleReporterFormat.exec(program.reporterOptions))) {
+      if (match.length !== 3) {
+        throw new Error("invalid multiple reporter options format '" + program.reporterOptions + "'");
+      }
+
+      var reporterName = match[1];
+      var reporterOptionStr = match[2];
+
+      reporterOptionStr.split(',').forEach(function(reporterOpt) {
+        var split = reporterOpt.indexOf('=');
+        if (split === -1) {
+          throw new Error("invalid reporter option '" + reporterOpt + "'");
         }
+        reporterOptions[reporterName] = reporterOptions[reporterName] || {};
```

Comment:
Cache the `options` object for this reporter outside the loop, and then you don't have to check it it's initialised to `{}` on each loop. Having a variable with a short name like `options` will also improve the legibility of the code.

After the loop you can assign back `options` to `reporterOptions[reporterName]`.